language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java

```diff
@@ -1624,6 +1624,28 @@ public static boolean isSelfClassSymbol(Symbol symbol, PositionedOperationContex
         return classSymbol.equals(varTypeSymbol);
     }
 
+    /**
+     * Check if the symbol is an object symbol with self as the name.
+     *
+     * @param symbol               Symbol
+     * @param nodeAtCursor         Node
+     * @return {@link Boolean} whether the symbol is a self object symbol.
+     */
+
+    public static boolean isSelfObjectSymbol(Symbol symbol, Node nodeAtCursor) {
+        Node currentNode = nodeAtCursor;
+        while (currentNode != null) {
+            if (currentNode.kind() == SyntaxKind.OBJECT_CONSTRUCTOR) {
+                break;
+            }
```

Comment:
What if we merge this condition (Actually the negation of it) with the while clause's condition?